components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/impl/OAuth2Authenticator.java

```diff
@@ -112,6 +115,18 @@ private boolean validateTokenAndScopes(Request request, ServiceMethodInfo servic
     }
 
     /**
+     * Extract the EndUsername from accessToken.
+     *
+     * @param accessToken the access token
+     * @return loggedInUser if the token is a valid token
+     */
+    private String getEndUserName(String accessToken) throws APIMgtSecurityException {
+        String loggedInUser;
+        loggedInUser = validateToken(accessToken).getEndUserName();
+        return loggedInUser.substring(0, loggedInUser.indexOf("@"));
```

Comment:
I think IS not give permission to create usernames with special characters such as @.but anyway I'll change that accordingly to last @.